### ISSUE TITLE
[Feature request] Connect directly through ADB

### ISSUE THREAD

**Issue by raffaem** [H-M-H/Weylus#76]
Currently I am suffering a lot of latency that renders Weylus not comfortable for my use case.

Would it be possible to connect through ADB directly (instead of TCP/IP) like [scrcpy](https://github.com/Genymobile/scrcpy) is doing?

**Comment by H-M-H** [comment]
Yes, I have already researched this and a direct usb connection is possible through the usb accessory mode on Android. But this will require writing an app as the browser can not directly connect via usb. Additionally most latency comes from video encoding and decoding. I have measured end to end latencies of about 20ms, more than half of that time was spend for encoding the video. I have not measured it but I suspect the latency of tcp over usb is not more than 1ms.

Perhaps it is possible to reduce latencies by skipping the video encoding and just sending raw pixeldata if the bandwidth is available. The question then is how efficient the browser is at drawing all these raw pixels.

**Comment by raffaem** [comment]
1. AFAIK [scrcpy](https://github.com/Genymobile/scrcpy) does not require any Android app

2. Can I change the compression settings? The resolution is very high but the latency is too much

**Comment by H-M-H** [comment]
scrcpy actually copies a program over to the android device via adb on startup, but that program doesn't need to display anything, so it's not a fully fledged app.

Changing the compression settings is not yet possible but another reason your device might be slow is that the frame rate your computer can encode with is too high for your tablet to keep up with, try setting the `Frame Interval` to something like 30ms.

**Comment by raffaem** [comment]
With this setting a bit better but still too high latency for my use case

**Comment by mukahraman** [comment]
Here is what I did,
installed `gnirehtet` which is a nice piece of software to reverse tether.
I am not sure this is needed but opened ports
```
adb reverse tcp:1701 tcp:1701
adb reverse tcp:9001 tcp:9001
```

Set ip to `127.0.0.1` on weylus.
The address is `10.0.0.2:1701` (might be different for you I'm not sure, check gnirehtet outputs)

When wifi was on, I couldn't connect FYI. Priority issues I suppose.

In my case, comparing over usb and wifi performed almost the same which is really interesting. I guess the bottleneck is video encoding/decoding.

**Comment by H-M-H** [comment]
Please try one of the latest builds, there is an option to lower the video resolution now. Like that you should be able to lower the latency considerably. I think the original issue can be considered as resolved with #71.
